src/neo4j_genai/neo4j_queries.py

```diff
@@ -56,9 +56,9 @@ def _get_hybrid_query() -> str:
 
 def _get_filtered_vector_query(
     filters: dict[str, Any],
-    node_label: str,
-    embedding_node_property: str,
-    embedding_dimension: int,
+    node_label: Optional[str],
```

Comment:
I think at this stage the parameters should not be optional anymore (None values are not managed inside this function). But if you had to add it, probably this is something we need to change in the code.

Reply:
This one is the golden crown. If we don't make them optional then there are some parts of the code that need to be changed because the code sends `None` in some cases :D